sycl/include/CL/sycl/id.hpp

```diff
@@ -16,7 +16,6 @@ namespace cl {
 namespace sycl {
 template <int dimensions> class range;
 template <int dimensions = 1> struct id : public detail::array<dimensions> {
-public:
```

Comment:
I agree that replacing `class` by `struct` is a simplification in the current situation.
But I wonder whether the original author did not mean instead having the next line with `base` private, so that `public:` was indeed on the wrong line...
Is there a reason to have `base` publicly visible?

Reply:
Well, the main goal of all those changes ("class" --> "struct") was the sync with SYCL SPEC and the wish to avoid build warnings like "this class was initially declared as struct".

In the additional change today I fixed 'id.hpp' and 'range.hpp' to hide 'base' inside private section.

Reply:
Hi Ronan, 
Please approve this PR if you agree with the additional changes.
Thank you,
Vyacheslav